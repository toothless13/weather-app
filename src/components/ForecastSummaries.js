import ForecastSummary from "./ForecastSummary";

const ForecastSummaries = ({ forecasts }) => {
  // const { date, description, icon, temperature} = props;
  // console.log(forecasts);
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => {
        const { date, description, icon, temperature } = forecast;
        return (
          <ForecastSummary
            key={date}
            date={date}
            description={description}
            icon={icon}
            temperature={temperature}
          />
        );
      })}
    </div>
  );
};

export default ForecastSummaries;
