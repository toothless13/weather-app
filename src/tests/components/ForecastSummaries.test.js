// eslint-disable-next-line no-unused-vars
import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries component", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 2222222,
      description: "Stub description 2",
      icon: "stubIcon2",
      temperature: {
        min: 13,
        max: 23,
      },
    },
    {
      date: 3333333,
      description: "Stub description 3",
      icon: "stubIcon3",
      temperature: {
        min: 14,
        max: 24,
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />,
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(3);
  });
});
