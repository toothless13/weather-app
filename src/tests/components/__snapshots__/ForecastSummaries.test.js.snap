// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="icon showing weather code"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="icon showing weather code"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        23°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="icon showing weather code"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 3
      </div>
    </div>
  </div>
</DocumentFragment>
`;
